module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -158,15 +158,16 @@ namespace module::input {
             // Here is an explanation https://gamedev.stackexchange.com/a/157018
             // Here is another derivation https://fgiesen.wordpress.com/2012/08/24/quaternion-differentiation/
             // dq/dt = (1/2)*omega*Rwt
-            const Eigen::Quaternion<Scalar> dq_dt = Eigen::Quaternion<Scalar>(0.0,
-                                                                              newState.omegaTTt.x() * 0.5,
-                                                                              newState.omegaTTt.y() * 0.5,
-                                                                              newState.omegaTTt.z() * 0.5)
-                                                    * Rwt;
+            const Eigen::Quaternion<Scalar> dq_dt =
+                Eigen::Quaternion<Scalar>(Eigen::Matrix<Scalar, 4, 1>(newState.omegaTTt.x() * 0.5,
+                                                                      newState.omegaTTt.y() * 0.5,
+                                                                      newState.omegaTTt.z() * 0.5,
+                                                                      0.0))
+                * Rwt;
             // The change in the rotation is the derivative times the differential (which is the time-step)
-            const Eigen::Quaternion<Scalar> change = Eigen::Quaternion<Scalar>(deltaT * dq_dt.coeffs());
-            // We can add the change to the original, as long as our time step is small enough
-            newState.Rwt = Eigen::Quaternion<Scalar>(Rwt.coeffs() + change.coeffs()).normalized();
+            const auto change = deltaT * dq_dt.coeffs();
```

Comment:
Do not use auto when you are working with Eigen math operations.
https://eigen.tuxfamily.org/dox/TopicPitfalls.html
You fell for one of the classic blunders

Reply:
``` suggestion 
const Eigen::Matrix<double, 4, 1> change(deltaT * dq_dt.coeffs());
```

Reply:
Imma have to repost the meme :joy:

Reply:
Well that page is quite interesting. Especially the part about passing by value. that makes things a bit spicy